### PR TITLE
openssl-pkcs11 -- update to 0.4.12

### PR DESCRIPTION
**Pull request by tobiasb-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 
Update `openssl-pkcs11` to latest release.

###### Change Log  
- Updated to `openssl-pkcs11` version `0.4.12`, tweaking only slightly from `fedora 40`

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
- Tested locally in container
- Buddy Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=566834&view=results

### CODE REVIEW

File: SPECS/openssl-pkcs11/openssl-pkcs11.spec [hunk 1]
```diff
@@ -1,46 +1,60 @@
+%define fedora 41
```
Comment (corvus-callidus) [inline SPECS/openssl-pkcs11/openssl-pkcs11.spec:1]: This feels weird. I know we're pulling an updated spec from fedora, but we're not using it for fedora. It's also making the conditionals further down the file less intuitive. I look at them and think "Oh, 'if fedora', this isn't fedora, don't need to pay attention to that" which is not actually the case.

Comment (anphel31) [reply SPECS/openssl-pkcs11/openssl-pkcs11.spec:1]: +1, this is a bit confusing.

Comment (tobiasb-ms) [reply SPECS/openssl-pkcs11/openssl-pkcs11.spec]: Looking at it again, I agree. Removed all this nonsense.

File: SPECS/openssl-pkcs11/openssl-pkcs11.spec [hunk 2]
```diff
@@ -92,13 +118,27 @@ make check %{?_smp_mflags} || { cat tests/*.log; false; }
...
+%if 0%{?fedora}
 %files -n libp11-devel
 %doc examples/ doc/api.out/html/
 %{_libdir}/libp11.so
 %{_libdir}/pkgconfig/libp11.pc
 %{_includedir}/*.h
+%endif
 
 %changelog
+* Thu Feb 08 2024 Jakub Jelen <jjelen@redhat.com> - 0.4.12-8
```
[the remark this answers is not in the document]
Comment (tobiasb-ms) [reply SPECS/openssl-pkcs11/openssl-pkcs11.spec]: done.

File: SPECS/openssl-pkcs11/openssl-pkcs11.spec [hunk 3]
```diff
@@ -1,40 +1,49 @@
...
+# unbreak operation when some other engine is present in openssl.cnf
+# https://github.com/OpenSC/libp11/pull/460
+# https://github.com/OpenSC/libp11/commit/feb22a66
+# 580c12b78b63d88010a6178d7c4c58186938c479
+# 74497e0fa5b69b15790d6697e1ebce13af842d4c
+Patch5:         openssl-pkcs11-ossl3.patch
+Patch6:         openssl-pkcs11-ec-copy.patch
+
+BuildRequires: make
+BuildRequires:  autoconf automake libtool
```
Comment (anphel31) [inline SPECS/openssl-pkcs11/openssl-pkcs11.spec:28]: nit: would prefer these on separate lines

**Review by anphel31 — APPROVED** [review]

File: SPECS/openssl-pkcs11/openssl-pkcs11.spec (hunk 3 above)
Comment (tobiasb-ms) [reply SPECS/openssl-pkcs11/openssl-pkcs11.spec:28]: agree but I figured I'd keep it this way in the interest of staying as close as reasonably possible to the upstream.

**Pull request merged**